util/strong_random.py

```diff
@@ -0,0 +1,137 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+"""Random number generators.
+
+A class for generating random numbers and permutations, by fetching entropy
+from a file.
+"""
+
+import random as rnd
+from math import ceil as _ceil
+from math import log as _log
+from warnings import warn as _warn
+
+import numpy as np
+
+
+class strong_random():
+
+    def __init__(self):
+        """Initialize an instance.
+
+        Entropy buffer is empty by default.
+        """
+
+        self.next = None
+        self.empty = True
+        self.buffer = np.array([])
+
+    def load(self, input_file):
+        """Load entropy buffer from an external file.
+
+        Currently only supports numpy array of 8-bit values.
+        """
+
+        self.buffer = np.load(input_file)
+        self.empty = False
+        self.next = 0
+
+    def size(self):
+        """ Size of the buffer.
+        """
+        return np.size(self.buffer)
+
+    def printstatus(self):
+        """Print the status of the buffer and the number of remaining bytes.
+
+        May be useful for testing and debugging.
+        """
+
+        print(self.buffer)
+        if self.empty:
+            print('Buffer is empty.')
+        else:
+            print("Buffer remaining:", len(self.buffer), "bytes.")
+
+    def isempty(self):
+        """Checks if the buffer is empty.
+        """
+        return self.empty
```

Comment:
Maybe it would make sense to make this a calculation and replace usage of `self.empty` with this function? That would reduce the risk that someone forgets to update `self.empty` when modifying the buffer. At that point, you could remove `self.empty` entirely.
```suggestion
    def isempty(self):
        """Checks if the buffer is empty.
        """
        return (self.buffer.size == 0)
```

Reply:
That indeed makes more sense. It is changed now.